.github/workflows/data-loader-ci.yaml

```
@@ -0,0 +1,45 @@
+name: CI for CLI
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  data_loader_ci:
+    name: Build Data Loader CLI
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set version
+        id: version
+        run: |
+          VERSION=$(echo "${GITHUB_REF}" | sed -e "s#refs/tags/v##g")
+          echo "version=${VERSION}" >> $GITHUB_OUTPUT
+
+      - name: Set up JDK 8 and 11 (default 8)
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: |
+            11
+            8
+
+      - name: Cache Gradle packages
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
```

I may miss something, but if you don't require a custom caching logic, using the `setup-gradle` action is usually preferred for a grade build compared to using the `action/cache` action. 

```suggestion
      - name: Setup Gradle
      uses: gradle/actions/setup-gradle@v4
```

---

Updated